### PR TITLE
Retransmission manager chagne to unique_ptr

### PR DESCRIPTION
**Pull request by titusen**
As above

### CODE REVIEW

File: bftengine/src/bftengine/ReplicaImp.cpp [hunk 1]
```diff
@@ -4499,11 +4499,10 @@ ReplicaImp::ReplicaImp(bool firstTime,
...
       config_.getcVal(), config_.getfVal(), config_.getreplicaId(), getCurrentView(), primaryLastUsedSeqNum);
 
-  if (retransmissionsLogicEnabled)
-    retransmissionsManager =
-        new RetransmissionsManager(&internalThreadPool, &getIncomingMsgsStorage(), kWorkWindowSize, 0);
-  else
-    retransmissionsManager = nullptr;
+  if (retransmissionsLogicEnabled) {
+    retransmissionsManager.reset(
+        new RetransmissionsManager(&internalThreadPool, &getIncomingMsgsStorage(), kWorkWindowSize, 0));
```
Comment (glevkovich) [inline bftengine/src/bftengine/ReplicaImp.cpp]: The code  will work, but using  make_unique is the correct way here. reset() should be used if we suspect that a pointer to another object was there before. This is the ctor:
```
if (retransmissionsLogicEnabled) {
   retransmissionsManager = std::make_unique<RetransmissionsManager>(&internalThreadPool,&getIncomingMsgsStorage(), kWorkWindowSize, 0);
}
```

File: bftengine/src/bftengine/ReplicaImp.hpp [hunk 2]
```diff
@@ -89,7 +89,7 @@ class ReplicaImp : public InternalReplicaApi, public ReplicaForStateTransfer {
   concord::util::SimpleThreadPool internalThreadPool;  // TODO(GG): !!!! rename
 
   // retransmissions manager (can be disabled)
-  RetransmissionsManager* retransmissionsManager = nullptr;
+  unique_ptr<RetransmissionsManager> retransmissionsManager = nullptr;
```
Comment (glevkovich) [inline bftengine/src/bftengine/ReplicaImp.hpp]: nit: you can remove the nullptr.
The Initial state of a smart pointer is always null.

**Review by glevkovich — CHANGES_REQUESTED** [review]

File: bftengine/src/bftengine/ReplicaImp.hpp (hunk 2 above)
Comment (titusen) [reply bftengine/src/bftengine/ReplicaImp.hpp]: Done

File: bftengine/src/bftengine/ReplicaImp.cpp (hunk 1 above)
Comment (titusen) [reply bftengine/src/bftengine/ReplicaImp.cpp]: Done

**Review by glevkovich — APPROVED** [review]
LGTM!

**Pull request merged**